src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs

```diff
@@ -63,4 +64,48 @@ public async Task<ControllingMemberCrcApplicationCmdResp> CreateControllingMembe
         return new ControllingMemberCrcApplicationCmdResp((Guid)app.spd_applicationid, (Guid) contact.contactid);
     }
 
+    public async Task<ControllingMemberCrcApplicationCmdResp> SaveControllingMenberCrcApplicationAsync(SaveControllingMemberCrcAppCmd cmd, CancellationToken ct)
+    {
+        spd_application? app;
+        if (cmd.ControllingMemberCrcAppId != null)
+        {
+            app = _context.spd_applications
+                .Expand(a => a.spd_application_spd_licencecategory)
+                .Where(a => a.spd_applicationid == cmd.ControllingMemberCrcAppId).FirstOrDefault();
+            if (app == null)
+                throw new ArgumentException("invalid app id");
+            _mapper.Map<SaveControllingMemberCrcAppCmd, spd_application>(cmd, app);
+            app.spd_applicationid = (Guid)(cmd.ControllingMemberCrcAppId);
+            _context.UpdateObject(app);
+        }
+        else
+        {
+            app = _mapper.Map<spd_application>(cmd);
+            _context.AddTospd_applications(app);
+            var contact = _context.contacts.Where(l => l.contactid == cmd.ContactId).FirstOrDefault();
+            if (contact != null)
+            {
+                _context.SetLink(app, nameof(spd_application.spd_ApplicantId_contact), contact);
+            }
+        }
+        //TODO: what is the LinkServiceType??
+        SharedRepositoryFuncs.LinkServiceType(_context, cmd.WorkerLicenceTypeCode, app);
+   
+    
+        //TODO: what is the LinkTeam ??
+        await LinkTeam(DynamicsConstants.Licensing_Client_Service_Team_Guid, app, ct);
+        await _context.SaveChangesAsync();
+      
+
+        //TODO: do we have category codes in the crc controlling member process?
```

Comment:
there is no categoryCode for controllingMember, if it has, your payload would include that.